src/core/qgsdistancearea.cpp

```diff
@@ -537,6 +540,111 @@ double QgsDistanceArea::measureLine( const QgsPoint &p1, const QgsPoint &p2, Qgs
   return result;
 }
 
+double QgsDistanceArea::measureLineProjected( const QgsPoint &p1, double distance, double azimuth, QgsPoint *projected_point )
+{
+  double result = 0.0;
+  QgsPoint p2;
+  if ( geographic() )
```

Comment:
> The setting of 'mSemiMajor, mSemiMinor, mInvFlattening' must be done for this to work. The default constructor is Wsg84, but not initialized. Also for other functions such as setSourceCrs, these values are not being set when geographic.

They are set when the ellipsoid is set (setEllipsoid). So the default constructor results in a QgsDistanceArea which performs cartesian calculations only. setEllipsoid must be called before any ellipsoidal calculations are used.

I'm not sure on the background behind this decision, but let's not get this PR distracted by that discussion. Let's get this merged first and then raise the discussion on the list about whether QgsDistanceArea should be initialized on construction.